### PR TITLE
feat: support multiple deployment environments

### PR DESCRIPTION
**Pull request by vtsaplin**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by askayastha22 — COMMENTED** [review]
I couldn't get my deployment environment to work for some reason with the provided URL. I see the version currently deployed on QA when I launch the URL.

**Comment by vtsaplin** [comment]
> I couldn't get my deployment environment to work for some reason with the provided URL. I see the version currently deployed on QA when I launch the URL.

@askayastha22 This is expected. The routing code should be in the dev environment (and later QA), but it was replaced by the localization PR. Once it's merged, it will be in QA and everything will work fine.

File: scripts/deploy.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  if (WORKSPACE_PRODUCTION === workspaceName) {
+    return 'https://experience.adobe.com/#/aem/generate-variations/';
+  } else if (WORKSPACE_STAGE === workspaceName) {
+    return 'https://experience-stage.adobe.com/?shell_ims=prod#/aem/generate-variations/';
+  } else if (WORKSPACE_QA === workspaceName) {
+    return 'https://experience-qa.adobe.com/?shell_ims=prod#/aem/generate-variations/';
+  } else if (WORKSPACE_DEVELOPMENT === workspaceName) {
+    return 'https://experience-qa.adobe.com/?devMode=true&shell_source=dev&shell_ims=prod#/aem/generate-variations/';
+  }
+  return `https://experience-qa.adobe.com/?devMode=true&shell_source=dev&shell_ims=prod&app_workspace=${workspaceName}#/aem/generate-variations/`;
```
Comment (askayastha22) [inline scripts/deploy.js:88]: `devMode=true` and `shell_source=dev` are unnecessary here for the QA URL.

**Review by askayastha22 — CHANGES_REQUESTED** [review]
Left some comments.

File: scripts/deploy.js (hunk 1 above)
Comment (vtsaplin) [reply scripts/deploy.js:88]: I would keep it there as it is supposed to help with CSP: https://git.corp.adobe.com/pages/exc/unified-shell-docs/docs/development/query-params#devmode

**Review by askayastha22 — APPROVED** [review]
👍

**Pull request merged**